src/net/socks5.cpp

```diff
@@ -10,14 +10,15 @@ namespace net {
 
 Socks5::Socks5(Settings s, Logger *lp, Poller *poller)
     : Emitter(lp), settings(s),
-      conn(settings["family"].c_str(), settings["socks5_address"].c_str(),
-           settings["socks5_port"].c_str(), lp, poller),
-      proxy_address(settings["socks5_address"]),
-      proxy_port(settings["socks5_port"]) {
+      conn(settings["family"].as<std::string>().c_str(), 
+           settings["socks5_address"].as<std::string>().c_str(),
+           settings["socks5_port"].as<std::string>().c_str(), lp, poller),
+      proxy_address(settings["socks5_address"].as<std::string>()),
+      proxy_port(settings["socks5_port"].as<std::string>()) {
 
     logger->debug("socks5: connecting to Tor at %s:%s",
-                  settings["socks5_address"].c_str(),
-                  settings["socks5_port"].c_str());
+                  settings["socks5_address"].as<std::string>().c_str(),
+                  settings["socks5_port"].as<std::string>().c_str());
```

Comment:
The use case whether one wants to convert a `Item` to the corresponding C string is quite common. Since we're in C++ and in C++ it's custom not to automatically convert `std::string` to `const char *`, I'd say it would be good to have a `Item::c_str()` method that just takes the `c_str()` of the internal string and returns it.

Does that makes sense?